File: web-admin-front/src/mock/cinema.ts
import type { CinemaItem } from '../types/cinema';

export const mockCinemas: CinemaItem[] = [
  {
    id: 1, name: '万达影城', branch: 'IMAX店',
    address: '北京市朝阳区建国路88号万达广场5层',
    longitude: 116.4731, latitude: 39.9087,
    facilities: ['IMAX', '杜比', '4DX', 'VIP厅'],
    rating: 8.5, phone: '010-88886666',
    status: 'active', hallCount: 6,
  },
  {
    id: 2, name: 'CGV影城', branch: '颐堤港店',
    address: '北京市朝阳区酒仙桥路18号颐堤港3层',
    longitude: 116.4905, latitude: 39.9653,
    facilities: ['IMAX', 'Dolby Atmos', '儿童厅'],
    rating: 8.2, phone: '010-66668888',
    status: 'active', hallCount: 5,
  },
  {
    id: 3, name: '大地影院', branch: '望京店',
    address: '北京市朝阳区望京街9号望京国际商业中心4层',
    longitude: 116.4815, latitude: 39.9904,
    facilities: ['巨幕厅', 'Reald 3D'],
    rating: 7.8, phone: '010-55551234',
    status: 'active', hallCount: 4,
  },
  {
    id: 4, name: '百老汇影城', branch: '三里屯店',
    address: '北京市朝阳区三里屯太古里南区B1',
    longitude: 116.4551, latitude: 39.9322,
    facilities: ['杜比', 'VIP厅'],
    rating: 8.9, phone: null,
    status: 'active', hallCount: 3,
  },
  {
    id: 5, name: '橙天嘉禾影城', branch: '通州店',
    address: '北京市通州区新华西街58号万达广场4层',
    longitude: 116.6633, latitude: 39.9042,
    facilities: ['4DX', '巨幕厅'],
    rating: null, phone: '010-77778888',
    status: 'closed', hallCount: 4,
  },
];

File: ticket-service/src/main/java/org/dherhf/cinema/entity/Cinema.java
package org.dherhf.cinema.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 影院实体,映射 {@code cinema} 表。
 * <p>
 * 影院 ID 使用雪花算法生成,设施字段通过 {@link JacksonTypeHandler} JSON 序列化存储,
 * 支持逻辑删除。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName(value = "cinema", autoResultMap = true)
public class Cinema {

    /** 影院 ID（雪花算法生成） */
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /** 影院名称 */
    private String name;

    /** 影院地址 */
    private String address;

    /** 经度 */
    private BigDecimal longitude;

    /** 纬度 */
    private BigDecimal latitude;

    /** 设施列表（如 IMAX、杜比等）,以 JSON 形式存储 */
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> facilities;

    /** 评分 */
    private BigDecimal rating;

    /** 联系电话 */
    private String phone;

    /** 状态：1-营业,0-停业 */
    private Integer status;

    /** 逻辑删除标记：0-未删除,1-已删除 */
    @TableLogic
    private Integer deleted;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}